tests/jballerina-unit-test/src/test/resources/test-src/statements/assign/assign-stmt.bal

```diff
@@ -86,3 +86,41 @@ public function restActionResultAssignment() returns [int, int, string, string,
     var error(r2, failedAttempts = failedAttempts) = c->foo3();
     return [a, b, d, r, r2, <int>failedAttempts];
 }
+
+function testAssignErrorArrayToAnyArray() {
+    string errorMessage = "Test Error";
+    error testError = error("Test Error");
+    error[] errorArray = [testError];
+    any anyArray = errorArray;
+    error[] errorArrayBack = <error[]>anyArray;
+    assertEquality(errorMessage, errorArrayBack[0].reason());
+}
+
+function testAssignIntArrayToJson() {
+    int[*] intArray = [1, 2];
+    json jsonObjectOfIntArray = intArray;
```

Comment:
This variable name does not make sense IMO.

Shall we rename it to `jsonVar` or `jsonArray`?